feature_engine/selection/drop_psi_features.py

```diff
@@ -419,6 +434,16 @@ def fit(self, X: pd.DataFrame, y: pd.Series = None):
             self.psi_values_[feature] = np.sum(
                 (test_distrib - basis_distrib) * np.log(test_distrib / basis_distrib)
             )
+
+            # Determine the appropriate threshold
+            if self.threshold == "auto":
```

Comment:
Hi @solegalli,

Let me try to address all the comments (including the other ones you left) in this one reply: 

> Hey @dlaprins
> 
> Thanks for the code changes. I went over the code and discussion, here are some more thoughts:
> 
>     * [ ]  we are now not capturing numerical and categorical variables in new attributes, that's great. Later on in the logic we are still looping over self.num_variables_ even though it does not exist any more. Could you please fix?

An oversight indeed.

>     * [ ]  It is not necessary to loop over self.variables_ we can remove the nested looping.

I'm a bit confused here. What do you mean by nested looping? As it stands, I believe there is a single for-loop over self.variables and only an if-statement inside of that for-loop, no other for-loops (for what it's worth: the for-loop over self.variables is necessary due to the dropna(), I tried vectorizing the for-loop before I noticed this and got wrong results). 

 >     * [ ]  n_bins is being reassigned for each feature, this is not necessary for numerical, only for categorical. Could we update?

The only way I see to update this is to use 2 loops, one over categoricals and one over numericals. Then n_bins doesn't have to be checked for each feature in the numerical loop. Is that what you have in mind? 

>     * [ ]  the threshold is calculated once for all variables, I think the idea is to use the same threshold for numerical, and a changing threshold for categorical. Could we reflect this in the code?

The threshold calculation is inside the loop over self.variables. Similar to n_bins, we can have a single numerical threshold outside of a for-loop over numericals, while for categoricals the threshold has to be calculated inside of the for-loop. 

>     * [ ]  The question still remains: should we expose p_values?

I believe we should. As a user I would find it rather unpleasant if I cannot set my own p-value threshold for a statistical test. It's fine if a user does not know what p_value they should set, as there's a default value for the p-value that they can use. 

>     * [ ]  it would be great if we could add the threshold calculation formula in the init docstrings as well.

 Will do

>     * [ ]  @glevv mentioned that the calculation would not work for this example: current implementation will give erroneous results if you have some differences in categories in the same columns (cat_1_train have categories 'A', 'B' and 'C', and cat_1_test have categories 'A', 'B' and 'D'). Could we add that as a test and make sure it works somehow? or raises an exception to alert users on what to do?

Is this really the case? I believe this is tested for by Claudio's test (see line 35, 85, 100 of test_drop_high_psi_features.py): "drift_cat_1" is exactly such that basis_test contains only "A" while test_df contains only "B". This is also why I got errors before modifying _observation_frequency_per_bin() when I changed the "drift_cat_1" feature in the test to be a categorical feature. Please correct me if I'm wrong here. 

> 
> Thank you for the great work!

Thank you for the suggestions on how to improve the code. 

I have left the loop over self.variables as it is for now, but if you would prefer to have 2 loops instead (or if I'm just not understanding you properly), let me know and I'll correct it.

Reply:
Hi @dlaprins yes, that was what I meant, 2 for loops instead of the for and the ifs. Sorry I wasn't clear.

Thank you for answering all the points!